### PR TITLE
Test for [NGMimeMessageGenerator generateDataForHeaderField:value:]

### PR DESCRIPTION
**Pull request by javierag**

### CODE REVIEW

File: Tests/Unit/SOGoTest.m [hunk 1]
```diff
@@ -185,4 +185,65 @@ - (BOOL) run
...
+  } else {
+    chrStr = @"[none]";
+  }
+
+  return chrStr;
+}
+
+/*
+  Returns a string with a very verbose diff
+*/
```
Comment (blaxter) [inline Tests/Unit/SOGoTest.m]: I don't know what this method does with this description

**Comment by javierag** [comment]
Expanded the  description for [SOGoTest diffForString:andString method]

File: Tests/Unit/SOGoTest.m [hunk 2]
```diff
@@ -185,4 +185,76 @@ - (BOOL) run
   return YES;
 }
 
+/* Helper function for diffForString:andString */
+NSString *_stringFoCharacterAtIndex(NSUInteger index, NSString *str, NSUInteger length)
```
Comment (sixstone-qq) [inline Tests/Unit/SOGoTest.m]: Typo: `_stringForCharacterAtIndex`

File: Tests/Unit/SOGoTest.h [hunk 3]
```diff
@@ -52,6 +54,9 @@
 
 - (BOOL) run;
 
+- (NSString*) diffForString: (NSString*) str1
+                  andString: (NSString*) str2;
```
Comment (sixstone-qq) [inline Tests/Unit/SOGoTest.h]: As we are creating a new `NSString`, according to Objective-C standard, we should start the message with string prefix, as developers must know this returned value is autoreleased and no further memory management action is required. So what about changing to `stringFromDiffBetween: str1 and: str2`?

**Comment by javierag** [comment]
Changed method names following Sixstone's tips

**Comment by blaxter** [comment]
:+1: probably we should integrate this on jenkins, someday...

File: Tests/Unit/SOGoTest.m [hunk 4]
```diff
@@ -185,4 +185,76 @@ - (BOOL) run
   return YES;
 }
 
+/* Helper function for diffForString:andString */
+NSString *_stringForCharacterAtIndex(NSUInteger index, NSString *str, NSUInteger length)
+{
+  NSString *chrStr;
+  unichar chr;
+  if (index < length) {
```
Comment (sixstone-qq) [inline Tests/Unit/SOGoTest.m]: Incorrect curly brace layout. According to SOGo, it should be:
```Objective-C
if (index < length)
  {
```

File: Tests/Unit/SOGoTest.m [hunk 5]
```diff
@@ -185,4 +185,76 @@ - (BOOL) run
...
+    else
+      {
+        if (chr == 10) 
+          chrStr = @"[NL]";
+        else if (chr == 0)
+          chrStr = @"[\0]";
+        else    
+          chrStr = [NSString stringWithFormat: @"[NP: %u]", chr];
+      }
+  } else {
```
Comment (sixstone-qq) [inline Tests/Unit/SOGoTest.m]: The same applies here.

**Comment by javierag** [comment]
Adjusted curl braces layout

**Comment by sixstone-qq** [comment]
:+1: and tested.

**Pull request merged**